src/lib/api/base.repo.spec.ts

```diff
@@ -0,0 +1,86 @@
+import anyTest, { TestInterface } from 'ava';
+import axios from 'axios';
+import MockAdapter from 'axios-mock-adapter';
+
+import { PermanentApiData } from '../model';
+
+import { BaseRepo, PermanentApiRequestI } from './base.repo';
+import { CsrfStore } from './csrf';
+
+const test = anyTest as TestInterface<{
+  baseRepo: BaseRepo;
+  csrfStore: CsrfStore;
+  mockAxios: MockAdapter;
+}>;
+const baseUrl = 'http://test.com';
+const apiKey = 'apiKey';
+
+test.beforeEach('New BaseRepo', (t) => {
+  const csrfStore = new CsrfStore();
+  const axiosInstance = axios.create({
+    baseURL: baseUrl,
+  });
+
+  const mockAxios = new MockAdapter(axiosInstance);
+
+  t.context = {
+    csrfStore,
+    mockAxios,
+    baseRepo: new BaseRepo({
+      csrfStore,
+      axiosInstance,
+      apiKey,
+    }),
+  };
+});
+
+test('requests are made using API key', async (t) => {
+  const endpoint = '/endpoint';
+
+  t.context.mockAxios.onPost(`${baseUrl}${endpoint}`).replyOnce((config) => {
+    const requestData = JSON.parse(config.data) as PermanentApiRequestI;
+    t.is(requestData.RequestVO.apiKey, apiKey);
+    return [200, {}];
+  });
+
+  await t.context.baseRepo.request(endpoint);
+});
+
+test('requests are made using csrf from CsrfStore', async (t) => {
+  const endpoint = '/endpoint';
+
+  t.context.csrfStore.setCsrf('testCsrf');
+  t.context.mockAxios.onPost(`${baseUrl}${endpoint}`).replyOnce((config) => {
+    const requestData = JSON.parse(config.data) as PermanentApiRequestI;
+    t.is(requestData.RequestVO.csrf, t.context.csrfStore.getCsrf());
+    return [200, {}];
+  });
+
+  await t.context.baseRepo.request(endpoint);
+});
+
+test('CsrfStore is updated with csrf from response', async (t) => {
+  const newCsrf = 'newcsrf';
+  const endpoint = '/endpoint';
+  t.context.csrfStore.setCsrf('oogabooga');
```

Comment:
Might be clearer to use a matching descriptive name:

```suggestion
  t.context.csrfStore.setCsrf('oldcsrf');
```

(Me, in this review, eliminating all fun from the codebase :sweat_smile: )

![Oogie Boogie](https://user-images.githubusercontent.com/1494855/96660964-dc7be680-12ff-11eb-9f84-80c3e3e88a99.png)